File: PROGRAMMERS/약수의 개수와 덧셈.py
# left = 13
left = 24
# right = 17
right = 27

def solution(left, right):
    answer = 0
    for x in range(left, right + 1):
        cnt = 0
        for j in range(1, x + 1):
            if x % j == 0:
                cnt += 1
        if cnt % 2 == 0:
            answer += x
        else:
            answer -= x
    return answer

print(solution(left,right))

File: PROGRAMMERS/JAVA/src/Lv_1/약수의개수와덧셈.java
package Lv_1;

import java.util.ArrayList;
import java.util.List;

public class 약수의개수와덧셈 {
    public int solution(int left, int right) {
        int answer = 0;
        List<Integer> tmp = new ArrayList<>();
        for (int i = left; i <= right; i++) {
            for (int j = 1; j <= i; j++) {
                if (i % j == 0) {
                    tmp.add(i);
                }
            }
            if (tmp.size() % 2 == 0) {
                answer += i;
            } else if (tmp.size() % 2 != 0) {
                answer -= i;
            }
            tmp.clear();
        }
        return answer;
    }
}
